src/driver.ml

```diff
@@ -169,6 +178,7 @@ module Transform = struct
         lint_intf;
         instrument;
         registered_at = caller_id;
+        kind = WholeFileTransform;
```

Comment:
What do you mean with `WholeFileTransform` here? It is also possible to register just one context-free rule.

Reply:
> Indeed, the current type Transform.t is not ensuring any safety: there can be transforms of kind Linters that contain no linter but an impl...

:heavy_plus_sign: 

> think this leads to errors, and I think it actually does in this particular case: when registering an impl together with an instrumentation, the impl function is never called... (I'll come back with more details on this bug!)

True. Good catch!